src/h5web/visualizations/shared/utils.ts

```diff
@@ -112,3 +112,15 @@ export function getTicksProp(
     ? { tickValues: getIntegerTicks(visibleDomain, numTicks) }
     : { numTicks };
 }
+
+export function assertNumOrStr(val: unknown): asserts val is number | string {
+  if (typeof val !== 'number' && typeof val !== 'string') {
+    throw new Error('Expected number or string');
+  }
+}
```

Comment:
Why not use the `ScalarValue` model here ?

Reply:
Just wanted it to be generic. To be perfectly honest, I don't think `ScalarValue` and `DataArray` are worth keeping. `MatrixVis` accepts a prop of type `ndarray<number | string>`, so I feel like sharing prop types between visualizations no longer makes much sense. Are you happy for me to remove them?

Reply:
I am all right with removing `ScalarValue` but I would keep `DataArray` for now as the single source of truth for the typing. I indeed foresee that the type `ndarray<number>` will change with improvements of #178.